pkg/receive/multitsdb.go

```diff
@@ -576,6 +604,42 @@ func (t *MultiTSDB) TenantAppendable(tenantID string) (Appendable, error) {
 	return tenant.readyStorage(), nil
 }
 
+// TODO: Handle case of one tenant in multiple hashrings
+func (t *MultiTSDB) SetHashringConfig(cfg []HashringConfig) error {
+	t.hashringConfigs = cfg
+
+	for _, hc := range t.hashringConfigs {
+		for _, tenantId := range hc.Tenants {
+			if t.tenants[tenantId] != nil {
+				lset := labelpb.ExtendSortedLabels(t.labels, labels.FromStrings(t.tenantLabelName, tenantId))
+				var elset labels.Labels
+				for _, el := range hc.ExternalLabels {
+					parts := strings.SplitN(el, "=", 2)
+					if len(parts) != 2 {
+						return errors.Errorf("unrecognized label %q", el)
+					}
+					if !model.LabelName.IsValid(model.LabelName(parts[0])) {
+						return errors.Errorf("unsupported format for label %s", el)
+					}
+					elset = append(elset, labels.Label{Name: parts[0], Value: parts[1]})
+				}
+				sort.Sort(elset)
+				lset = labelpb.ExtendSortedLabels(lset, elset)
+
+				t.tenants[tenantId].ship.SetLabels(lset)
+				t.tenants[tenantId].storeTSDB.SetExtLset(lset)
+				t.tenants[tenantId].exemplarsTSDB.SetExtLabels(lset)
+			}
```

Comment:
@fpetkovski Does the labels updating logic look somewhat like this?

This doesn't work. When I checked locally I got this error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1b9f735]

goroutine 161 [running]:
github.com/thanos-io/thanos/pkg/shipper.(*Shipper).SetLabels(...)
	/home/admin1/go/src/github.com/thanos-io/thanos/pkg/shipper/shipper.go:123
github.com/thanos-io/thanos/pkg/receive.(*MultiTSDB).SetHashringConfig(0xc0004ee580, {0xc002aa8b00?, 0x1, 0xd?})
	/home/admin1/go/src/github.com/thanos-io/thanos/pkg/receive/multitsdb.go:632 +0x355
main.setupHashring.func5()
	/home/admin1/go/src/github.com/thanos-io/thanos/cmd/thanos/receive.go:581 +0x232
github.com/oklog/run.(*Group).Run.func1({0xc000929800?, 0xc00028ffe0?})
	/home/admin1/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x2f
created by github.com/oklog/run.(*Group).Run
	/home/admin1/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x22a
```

But at least is this the right approach?